File: python-cli/src/convex_notes/handlers/delete.py
"""Delete note handler."""

from rich.console import Console
from rich.prompt import Confirm, Prompt

from convex_notes.service import NoteService

console = Console()


def execute(service: NoteService) -> None:
    """Execute the delete note command."""
    console.print("\n[bold red]🗑️ Delete a note[/bold red]")

    notes = service.list()

    if not notes:
        console.print("[yellow]📭 No notes to delete.[/yellow]")
        return

    # Display notes for selection
    console.print("\nSelect a note to delete:")
    for i, note in enumerate(notes, 1):
        console.print(f"  [cyan]{i}[/cyan]. {note.title}")

    try:
        choice = int(Prompt.ask("\n[cyan]Enter number[/cyan]")) - 1
        if choice < 0 or choice >= len(notes):
            console.print("[red]❌ Invalid selection.[/red]")
            return
    except ValueError:
        console.print("[red]❌ Invalid input.[/red]")
        return

    selected = notes[choice]

    if not Confirm.ask(f"Are you sure you want to delete '[yellow]{selected.title}[/yellow]'?"):
        console.print("[yellow]❌ Cancelled.[/yellow]")
        return

    try:
        service.delete(selected.id)
        console.print(f"\n[green]✅[/green] Note '[strike red]{selected.title}[/strike red]' deleted.")
    except Exception as e:
        console.print(f"[red]❌ Failed to delete note:[/red] {e}")

File: backend/crates/cli/src/handlers/delete.rs
//! Delete note handler.

use anyhow::Result;
use colored::Colorize;
use dialoguer::{Confirm, Select, theme::ColorfulTheme};

use db::{NoteRepository, NoteService};

/// Execute the delete note command.
pub async fn execute<R: NoteRepository>(service: &NoteService<R>) -> Result<()> {
    println!("\n{}", "🗑️ Delete a note".bright_red().bold());

    let notes = service.list().await?;

    if notes.is_empty() {
        println!("{}", "📭 No notes to delete.".bright_yellow());
        return Ok(());
    }

    let note_titles: Vec<&str> = notes.iter().map(|n| n.title.as_str()).collect();

    let selection = Select::with_theme(&ColorfulTheme::default())
        .with_prompt("Select a note to delete")
        .items(&note_titles)
        .interact()?;

    let selected = &notes[selection];

    let confirm = Confirm::with_theme(&ColorfulTheme::default())
        .with_prompt(format!(
            "Are you sure you want to delete '{}'?",
            selected.title.bright_yellow()
        ))
        .default(false)
        .interact()?;

    if !confirm {
        println!("{}", "❌ Cancelled.".bright_yellow());
        return Ok(());
    }

    match service.delete(&selected.id).await {
        Ok(_) => println!(
            "\n{} Note '{}' deleted.",
            "✅".bright_green(),
            selected.title.bright_red().strikethrough()
        ),
        Err(e) => println!("{} {}", "❌ Failed to delete note:".bright_red(), e),
    }

    Ok(())
}
